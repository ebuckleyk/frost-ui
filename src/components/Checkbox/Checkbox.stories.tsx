import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Checkbox } from './Checkbox';
import { Card } from '../Card';

function CheckboxDemo() {
  return (
    <Card className="flex items-center space-x-2 p-5">
      <Checkbox id="terms" />
      <label
        htmlFor="terms"
        className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
      >
        Accept terms and conditions
      </label>
    </Card>
  );
}
type ComponentType = React.ComponentProps<typeof Checkbox>;
const meta: Meta<ComponentType> = {
  component: Checkbox,
  render: CheckboxDemo,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
