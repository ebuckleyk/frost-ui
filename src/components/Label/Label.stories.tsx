import { Meta, StoryObj } from '@storybook/react';
import { Label } from './Label';

type ComponentType = typeof Label;
const meta: Meta<ComponentType> = {
  component: Label,
};

export default meta;

type Story = StoryObj<ComponentType>;

export const Default: Story = {
  args: {},
};
