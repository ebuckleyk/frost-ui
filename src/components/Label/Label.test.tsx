import * as React from 'react';
import { render } from '@testing-library/react';

import { Checkbox } from '../Checkbox';
import { Label } from './Label';

const Component = () => {
  return (
    <div className="flex items-center space-x-2">
      <Checkbox id="terms" />
      <Label htmlFor="terms">Accept terms and conditions</Label>
    </div>
  );
};

describe('Label', () => {
  it('should render Label component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
