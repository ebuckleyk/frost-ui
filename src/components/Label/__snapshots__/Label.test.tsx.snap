// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > should render Label component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center space-x-2"
      >
        <button
          aria-checked="false"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          data-state="unchecked"
          id="terms"
          role="checkbox"
          type="button"
          value="on"
        />
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="terms"
        >
          Accept terms and conditions
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center space-x-2"
    >
      <button
        aria-checked="false"
        class="peer h-4 w-4 shrink-0 rounded-sm border border-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
        data-state="unchecked"
        id="terms"
        role="checkbox"
        type="button"
        value="on"
      />
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="terms"
      >
        Accept terms and conditions
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
